### PR TITLE
refactor(types): doc & refine ToBinary/Text

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
* **#17690** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17690?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17697** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17697?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @xxchan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/connector/src/parser/mysql.rs [hunk 1]
```diff
@@ -149,7 +149,7 @@ mod tests {
     use mysql_async::Row as MySqlRow;
     use risingwave_common::catalog::{Field, Schema};
     use risingwave_common::row::Row;
-    use risingwave_common::types::{DataType, ToText};
+    use risingwave_common::types::DataType;
```
Comment (xxchan) [inline src/connector/src/parser/mysql.rs:152]: This change was made during browsing all references of `ToText`. Don't think it's worth exposing it here.

File: src/common/src/types/to_binary.rs [hunk 2]
```diff
@@ -30,14 +33,15 @@ pub enum ToBinaryError {
 
 pub type Result<T> = std::result::Result<T, ToBinaryError>;
 
-// Used to convert ScalarRef to text format
```
Comment (xiangjinwu) [inline src/common/src/types/to_binary.rs:33]: 🤦

File: src/common/src/types/to_text.rs [hunk 3]
```diff
@@ -18,7 +18,14 @@ use std::num::FpCategory;
 use super::{DataType, DatumRef, ScalarRefImpl};
 use crate::dispatch_scalar_ref_variants;
 
-// Used to convert ScalarRef to text format
+/// Converts `ScalarRef` to text format.
+/// This is the implementation for casting to varchar, and pgwire "TEXT" format.
```
Comment (xiangjinwu) [inline src/common/src/types/to_text.rs:22]: > This is the implementation for casting to varchar

This is not true for a few cases. The TEXT format for boolean is `t` / `f` while they cast to varchar `true` / `false`.

https://github.com/postgres/postgres/blob/REL_16_3/src/include/catalog/pg_cast.dat#L438-L439

https://www.postgresql.org/docs/16/sql-createcast.html#:~:text=A%20small%20number%20of%20the%20built%2Din%20types%20do%20indeed%20have%20different%20behaviors%20for%20conversions%2C%20mostly%20because%20of%20requirements%20of%20the%20SQL%20standard.

**Review by xiangjinwu — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**